src/py/mat3ra/made/tools/build/interface/builders.py

```diff
@@ -20,113 +20,101 @@
-class SimpleInterfaceBuilderParameters(InterfaceBuilderParameters):
-    scale_film: bool = True  # Whether to scale the film to match the substrate
-    create_slabs: bool = True  # Whether to create slabs from the configurations or use the bulk
-
-
-class SimpleInterfaceBuilder(ConvertGeneratedItemsASEAtomsMixin, InterfaceBuilder):
+class InterfaceBuilder(StackBuilder2Components):
     """
     Creates matching interface between substrate and film by straining the film to match the substrate.
     """
 
-    _BuildParametersType = SimpleInterfaceBuilderParameters
-    _DefaultBuildParameters = SimpleInterfaceBuilderParameters(scale_film=True)
-    _GeneratedItemType: type(ASEAtoms) = ASEAtoms  # type: ignore
-
-    def __preprocess_slab_configuration(self, configuration: SlabConfiguration, create_slabs=False) -> ASEAtoms:
-        slab = create_slab(configuration) if create_slabs else configuration.bulk
-        ase_slab = to_ase(slab)
-
-        niggli_reduce(ase_slab)
-        return ase_slab
-
-    @staticmethod
-    def __combine_two_slabs_ase(substrate_slab_ase: ASEAtoms, film_slab_ase: ASEAtoms, distance_z: float) -> ASEAtoms:
-        total_z_height = substrate_slab_ase.cell[2][2] + film_slab_ase.cell[2][2] + distance_z
-        substrate_slab_ase.cell[2][2] = total_z_height
-        film_slab_ase.cell[2][2] = total_z_height
-        max_z_substrate = max(substrate_slab_ase.positions[:, 2])
-        min_z_film = min(film_slab_ase.positions[:, 2])
-        shift_z = max_z_substrate - min_z_film + distance_z
-
-        film_slab_ase.translate([0, 0, shift_z])
-
-        return substrate_slab_ase + film_slab_ase
-
-    @staticmethod
-    def __add_vacuum_along_c_ase(interface_ase: ASEAtoms, vacuum: float) -> ASEAtoms:
-        cell_c_with_vacuum = max(interface_ase.positions[:, 2]) + vacuum
-        interface_ase.cell[2, 2] = cell_c_with_vacuum
-        return interface_ase
-
-    def _generate(self, configuration: InterfaceBuilder._ConfigurationType) -> List[_GeneratedItemType]:  # type: ignore
-        film_slab_ase = self.__preprocess_slab_configuration(
-            configuration.film_configuration,
-            create_slabs=self.build_parameters.create_slabs,
+    _ConfigurationType = InterfaceConfiguration
+    _BuildParametersType = InterfaceBuilderParameters
+    _GeneratedItemType: Type[Material] = Material
+
+    def _configuration_to_material(self, configuration_or_material: Any) -> Material:
+        if isinstance(configuration_or_material, Material):
+            return configuration_or_material
+
+        if isinstance(configuration_or_material, SlabStrainedSupercellConfiguration):
+            builder = SlabStrainedSupercellBuilder()
+            return builder.get_material(configuration_or_material)
+
+        return super()._configuration_to_material(configuration_or_material)
+
+    def _get_labeled_substrate_material(self, configuration: InterfaceConfiguration) -> Material:
```

Comment:
Same as above:  _get _labeled_component_material(component_configuration, label) then reuse it. Also - this functionality can be in `StackBuilder2Components`